src/main/java/picard/illumina/IlluminaBasecallsToFastq.java

```diff
@@ -201,18 +199,6 @@
     @Option(doc = "Use the new converter", optional = true)
     public boolean USE_NEW_CONVERTER = false;
 
-    @Option(doc = "Maximum mismatches for a barcode to be considered a match.")
-    public int MAX_MISMATCHES = 1;
-
-    @Option(doc = "Minimum difference between number of mismatches in the best and second best barcodes for a barcode to be considered a match.")
-    public int MIN_MISMATCH_DELTA = 1;
-
-    @Option(doc = "Maximum allowable number of no-calls in a barcode read before it is considered unmatchable.")
-    public int MAX_NO_CALLS = 2;
-
-    @Option(doc = "Per-barcode and per-lane metrics written to this file.", shortName = StandardOptionDefinitions.METRICS_FILE_SHORT_NAME, optional = true)
```

Comment:
Could this be deprecated first in one release, then removed?  Perhaps remove the functionality and add a custom command line validation to error when it is specified?  Seems like people may be using this.